src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorWorkspaceListener.cs

```diff
@@ -150,30 +118,43 @@ private void EnqueueUpdate(Project? project)
         }
 
         var projectId = project.Id;
-        if (_workQueues.TryGetValue(projectId, out var scheduler))
+        _workQueue.AddWork(projectId);
+    }
+
+    private async ValueTask UpdateCurrentProjectsAsync(ImmutableArray<ProjectId> projectIds, CancellationToken cancellationToken)
+    {
+        if (_workspace is null)
         {
-            _logger.LogTrace("{projectId} scheduling task due to workspace event", projectId);
+            _logger?.LogTrace("No workspace available when trying to write project info");
+            return;
+        }
 
-            scheduler.ScheduleAsyncTask(ct => SerializeProjectAsync(projectId, ct), CancellationToken.None);
+        if (_projectInfoFileName is null)
+        {
+            _logger?.LogTrace("Don't know what filename to use");
+            return;
         }
-    }
 
-    // Protected for testing
-    protected virtual Task SerializeProjectAsync(ProjectId projectId, CancellationToken ct)
-    {
-        if (_projectInfoFileName is null || _workspace is null)
+        var solution = _workspace.CurrentSolution;
+
+        foreach (var projectId in projectIds.Distinct())
         {
-            return Task.CompletedTask;
+            await SerializeProjectAsync(projectId, solution, cancellationToken).ConfigureAwait(false);
```

Comment:
This is looping through so it may call multiple times. Would it be like this? (if it exists)

```csharp
return ValueTask.WhenAll(projectIds.Select(pId => SeralizeProjectAsync(pId, solution, cancellationToken))
```

Reply:
Derp! I completely missed that this was inside a loop when looking at the diff. Ignore my suggestion here. 😄